### PR TITLE
#3: Placeholders in mitigations

### LINKED ISSUE #3: Placeholders in mitigations
## Situation

Sometimes mitigations benefit from dynamic content. To allow this, the API should be extended to support placeholders in mitigations.

## Acceptance Criteria

* Mitigations allow using placeholders.

### PR DESCRIPTION
**Pull request by redcatbear**
Closes #3.

### CODE REVIEW

File: doc/user_guide/user_guide.md [hunk 1]
```diff
@@ -49,8 +49,8 @@ As you can see in the example above, you can simply pass the error attributes as
 The predefined table keys are
 
 * `code`: machine-readable unique code (string)
-* `message`: Error description either as static string or as [format with placeholders](https://www.lua.org/manual/5.1/manual.html#pdf-string.format) (string)
-* `parameters`: table of parameter values be used to replace placeholders in the description
+* `message`: Error description either as static string or containing placeholders in double curly brackets.
+* `parameters`: table of parameter values be used to replace placeholders in the message and mitigations
 * `mitigations`: list of hints on how to fix the error (array of strings)
```
Comment (jakobbraun) [inline doc/user_guide/user_guide.md:54]: you could also mention the placeholders here

File: doc/user_guide/user_guide.md [hunk 2]
```diff
@@ -63,12 +63,41 @@ Strictly speaking it is a combination between the builder pattern and regular se
...
+As you can see from the examples the named parameters are placeholders that need to be enclosed in double curly brackets. You define the values in a table where the parameter names are the keys.
+
+You can use the same parameter replacement mechanism in mitigations that you saw earlier used in the message part.
+
+```lua
+local project_issue_url = "www.example.org/issues"
+-- ...
+local msg = exaerror:new({
+    message = "Unexpected error.",
+    mitigations = {"Please create an error report under {{url}}"},
```
Comment (jakobbraun) [inline doc/user_guide/user_guide.md:83]:
```suggestion
    mitigations = {"Please create an error report under {{url}}."},
```

File: doc/user_guide/user_guide.md [hunk 3]
```diff
@@ -63,12 +63,41 @@ Strictly speaking it is a combination between the builder pattern and regular se
...
+    parameters = {url = project_issue_url}
+})
+```
+
+If you want to add a description to your parameters, you need to invest just a little bit more effort:
+
+```lua
+local msg = exaerror:new({
+    message = "Unexpected error.",
+    mitigations = {"Please create an error report under {{url}}"},
```
Comment (jakobbraun) [inline doc/user_guide/user_guide.md:93]:
```suggestion
    mitigations = {"Please create an error report under {{url}}."},
```

**Review by jakobbraun — CHANGES_REQUESTED** [review]

File: doc/user_guide/user_guide.md (hunk 1 above)
Comment (redcatbear) [reply doc/user_guide/user_guide.md]: Adde.

**Review by jakobbraun — APPROVED** [review]

**Pull request merged**